modules/tensor_mechanics/src/materials/SingleVariableReturnMappingSolution.C

```diff
@@ -135,21 +142,15 @@ SingleVariableReturnMappingSolution::internalSolve(const Real effective_trial_st
                                                    Real & scalar,
                                                    std::stringstream * iter_output)
 {
-  scalar = 0.0;
-  Real scalar_old = 0.0;
+  scalar = initialGuess(effective_trial_stress);
+  Real scalar_old = scalar;
   Real scalar_increment = 0.0;
-  const Real min_permissible_scalar = 0.0;
+  const Real min_permissible_scalar = minimumPermissibleValue(effective_trial_stress);
   const Real max_permissible_scalar = maximumPermissibleValue(effective_trial_stress);
   Real scalar_upper_bound = max_permissible_scalar;
-  Real scalar_lower_bound = 0.0;
+  Real scalar_lower_bound = min_permissible_scalar;
   unsigned int it = 0;
 
-  if (effective_trial_stress == 0.0)
```

Comment:
@bwspenc This is the only place where effective_trial_stress is used in this class, and preventing complete removal. I wonder if it is even necessary to skip the iteration if stress is zero? Certainly in my models, even if stress is zero, I still need to run a newton iteration.

Reply:
@tophmatthews I believe we were having trouble with some models that divided by stress in the first timestep in the earlier model development.  With @bwspenc's changes in the iteration, we might not need to guard against it anymore.  I think the creep models are the most likely to demonstrate the problem, if it still exists

Reply:
Ah, that would make sense. I am guessing (maybe hoping) the tests would have picked up on this?

Reply:
I can't remember if the tests did catch the problem, but I'm fairly certain the bison example problems did; would you be willing to run those example problems with your branch to check please?

Reply:
sure. Remember which ones?

Reply:
the tensor mechanics 2D rodlet with finite strain. thank you!

Reply:
I'm guessing stress will only be zero during the initial timestep?

Reply:
yes - we'd see failure in the material solve immediately. If your change gets the example problem to _t_step = 2, we should be good to go. Thank you for checking this case!

Reply:
Good that will save time! All of the tensor mechanics 2DRZ problems run the first timestep fine, so it seems this may have been fixed.

Reply:
I'm good with removing this, but I'd like to also slightly re-work the logic around the call to `returnMappingSolve()` in `RadialReturnStressUpdate` that prevents a divide by zero as well.  I'd say let's change:
```
  // Use Newton iteration to determine the scalar effective inelastic strain increment
  Real scalar_effective_inelastic_strain = 0;
  returnMappingSolve(effective_trial_stress, scalar_effective_inelastic_strain, _console);

  if (scalar_effective_inelastic_strain != 0.0)
    inelastic_strain_increment = deviatoric_trial_stress *
                                 (1.5 * scalar_effective_inelastic_strain / effective_trial_stress);
  else
    inelastic_strain_increment.zero();
```
to:
```
  // Use Newton iteration to determine the scalar effective inelastic strain increment
  Real scalar_effective_inelastic_strain = 0;
  if (!MooseUtils::absoluteFuzzyEqual(effective_trial_stress, 0.0))
  {
    returnMappingSolve(effective_trial_stress, scalar_effective_inelastic_strain, _console);
    if (scalar_effective_inelastic_strain != 0.0)
      inelastic_strain_increment = deviatoric_trial_stress *
                                   (1.5 * scalar_effective_inelastic_strain / effective_trial_stress);
    else
      inelastic_strain_increment.zero();
  }
  else
    inelastic_strain_increment.zero();
```
...unless you really need to call into the Newton iteration once in your model.

Reply:
Yeah, I need to call Newton iteration at least once. There are contributions to the strain that are not related to the stress. We can add a flag to force at least one iteration, although it seems cleaner how it is.

Reply:
Oh I see, it still needs to protect the divide by zero...That's identified in #11295 as well

Reply:
Yeah, actually I'm good with your fix in `RadialReturnStressUpdate`. It moves the decision to do the newton solve back one step into the individual model. I have a corresponding model like `RadialReturnStressUpdate` where I can force the call. It'll also fix the divide by zero in #11295.